src/utils/calculateUtils.ts

```diff
@@ -0,0 +1,46 @@
+import { Panel } from "../types/types";
+
+/**
+ * 개별 패널의 style 객체를 계산하는 함수
+ *
+ * @param panel - 위치와 크기 정보를 담은 패널 객체
+ * @param unitWidth - 1 grid column 당 픽셀 너비
+ * @param rowHeight - 1 grid row 당 픽셀 높이
+ * @returns React에서 사용 가능한 CSSProperties 객체
+ */
+interface calculatePanelStyleProps {
+  panel: Panel,
+  unitWidth: number,
+  rowHeight: number
+}
+export function calculatePanelStyle({ panel, unitWidth, rowHeight }: calculatePanelStyleProps): React.CSSProperties {
+  const width = panel?.w || 1;
+  const height = panel?.h || 1;
+  const x = panel?.x || 0;
+  const y = panel?.y || 0;
+
+  return {
+    width: width * unitWidth,
+    height: height * rowHeight,
+    transform: `translate(${x * unitWidth}px, ${y * rowHeight}px)`,
+  };
+}
+
+/**
+ * 전체 grid container의 높이를 계산하는 함수
+ * 모든 패널의 y+h 중 최댓값을 기준으로 계산합니다.
+ *
+ * @param panels - 화면에 표시할 모든 패널 배열
+ * @param rowHeight - 1 row 당 픽셀 높이
+ * @returns 컨테이너 전체 높이 (px 단위)
+ */
+interface calculateContainerHeight {
+  panels: Panel[],
+  rowHeight: number
+}
```

Comment:
`@param`, `@returns`는 함수일 때 쓰는 것이고, 여기서는 아래와 같이 바꾸면 자동완성할 때 유용합니다.
```suggestion
/**
 * 전체 grid container의 높이를 계산하는 함수
 * 모든 패널의 y+h 중 최댓값을 기준으로 계산합니다.
 */
interface calculateContainerHeight {
  /** 화면에 표시할 모든 패널 배열 */
  panels: Panel[],
  /** row 당 픽셀 높이 */
  rowHeight: number
}
```

Reply:
기존에 함수 상단에 작성했던건데 type을 추가하면서 위치가 잘못되었던 것 같습니다. 
type의 경우 말씀해주신 사항 참고하여 작성하겠습니다!